cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/util/AbstractAppFabricHttpHandler.java

```diff
@@ -332,18 +332,27 @@ protected boolean respondIfElementNotFound(Throwable t, HttpResponder responder)
     return false;
   }
 
+  // TODO: refactor
   protected final void dataList(HttpRequest request, HttpResponder responder, Store store, DatasetFramework dsFramework,
-                                Data type, String namespace, String name, String appId) {
+                                Data type, String namespaceId, String name, String appId) {
     try {
       if ((name != null && name.isEmpty()) || (appId != null && appId.isEmpty())) {
         responder.sendString(HttpResponseStatus.BAD_REQUEST, "Empty name provided");
         return;
       }
 
-      Id.Program program = Id.Program.from(namespace, appId == null ? "" : appId, "");
-      String json = name != null ? getDataEntity(store, dsFramework, program, type, name) :
-        appId != null ? listDataEntitiesByApp(store, dsFramework, program, type)
```

Comment:
looks like you did the right thing, just wondering how we didn't catch this bug so far. am I missing something?